sklearn/conftest.py

```diff
@@ -37,6 +38,17 @@
     "fetch_rcv1_fxt": fetch_rcv1,
 }
 
+_SKIP32_MARK = pytest.mark.skipif(
+    environ.get("SKLEARN_SKIP_FLOAT32", "1") != "0",
+    reason="Set SKLEARN_SKIP_FLOAT32=0 to run float32 dtype tests",
+)
+
+
+# Global fixture
+@pytest.fixture(params=[pytest.param(np.float32, marks=_SKIP32_MARK), np.float64])
+def dtype(request):
```

Comment:
We use "dtype" as a `pytest.mark.parameterize" in many places where we actually want to test for all dtypes. Maybe we need to give this a more semantic name:

```suggestion
def global_dtype(request):
```

I am open for other suggestions.

Reply:
I agree that we need to make explicit that the current `dtype` is a global fixture. To me `pytest.fixture` comes in handy but with implicit definition which might be misleading for new comers reading the tests.

Could we come up with a convention for fixture definition and usage?

Though, I can not imagine a good option to make fixture explicit, I am OK using your proposed `global_` prefix as a convention for now.

Reply:
Addressed in 09ccf3196c948532974fb40d67a1f14813f7b06d.